core/services/workflows/syncer/workflow_registry.go

```diff
@@ -353,13 +407,192 @@ func (w *workflowRegistry) readRegistryEvents(ctx context.Context, don capabilit
 			}
 
 			for _, event := range events {
-				err := w.handler.Handle(ctx, event.Event)
+				w.eventCh <- event.Event
+			}
+		}
+	}
+}
+
+// syncUsingEventStrategy syncs workflow registry contract state by watching for events on the contract.
+// It first loads the workflow metadata from the contract state as WorkflowRegistered events,
+// and then starts a goroutine with one loop for handling the events.
+func (w *workflowRegistry) syncUsingEventStrategy(ctx context.Context, don capabilities.DON, reader ContractReader) {
+	w.lggr.Debugw("Loading initial workflows for DON", "DON", don.ID)
+
+	workflowMetadata, loadWorkflowsHead, err := w.getWorkflowMetadata(ctx, don, reader)
+	// TODO: if this fails, we should still start the event looping
+	if err != nil {
+		w.lggr.Errorw("failed to load initial workflows", "err", err)
+		return
+	}
+
+	w.lggr.Debugw("Rehydrating existing workflows", "len", len(workflowMetadata))
+	for _, workflow := range workflowMetadata {
+		toRegisteredEvent := WorkflowRegistryWorkflowRegisteredV1{
+			WorkflowID:    workflow.WorkflowID,
+			WorkflowOwner: workflow.Owner,
+			DonID:         workflow.DonID,
+			Status:        workflow.Status,
+			WorkflowName:  workflow.WorkflowName,
+			BinaryURL:     workflow.BinaryURL,
+			ConfigURL:     workflow.ConfigURL,
+			SecretsURL:    workflow.SecretsURL,
+		}
+		w.eventCh <- workflowAsEvent{
+			Data:      toRegisteredEvent,
+			EventType: WorkflowRegisteredEvent,
+		}
+		if err != nil {
+			w.lggr.Errorf("failed to handle workflow registration: %s", err)
+		}
+	}
+
+	// Poll for all workflow related events
+	ets := []WorkflowRegistryEventType{
+		ForceUpdateSecretsEvent,
+		WorkflowActivatedEvent,
+		WorkflowDeletedEvent,
+		WorkflowPausedEvent,
+		WorkflowRegisteredEvent,
+		WorkflowUpdatedEvent,
+	}
+
+	w.wg.Add(1)
+	defer w.wg.Done()
```

Comment:
You're not starting a goroutine here so you can remove

Instead you should do this logic where we spin up the goroutine here: https://github.com/smartcontractkit/chainlink/pull/16925/files